src/gennylib/test/PhaseLoop_test.cpp

```diff
@@ -119,6 +119,62 @@ TEST_CASE("Correctness for N milliseconds") {
     }
 }
 
+TEST_CASE("Correctness for N milliseconds with sleepNonBlocking") {
+    genny::metrics::Registry metrics;
+    genny::Orchestrator o{};
+    SECTION(
+        "Looping for 10 milliseconds takes between 10 and 11 millis and does 10 iterations with "
+        "sleepNonBlocking 1ms") {
+        v1::ActorPhase<int> loop{
+            o,
+            std::make_unique<v1::IterationChecker>(10_ots, nullopt, false, 0_ts, 0_ts, nullopt),
+            0};
+
+        auto start = chrono::system_clock::now();
+        int i = 0;
+        INFO("Before for range loop");
+        INFO(i);
+        for (auto _ : loop) {
+            ++i;
+            INFO("In loop");
+            loop.sleepNonBlocking(chrono::milliseconds(1));
+        }
+        INFO("After loop");
+        INFO(i);
+        auto elapsed =
+            chrono::duration_cast<chrono::milliseconds>(chrono::system_clock::now() - start)
+                .count();
+
+        REQUIRE(i >= 9);
+        // REQUIRE(i <= 11);
```

Comment:
This require is failing and I'm completely flummoxed. the thing is running a few hundred iterations. I tried to add some fudge factors wondering if I had a race, but it didn't help. It does run for the right amount of time though.